src/main/java/com/amazon/opendistroforelasticsearch/sql/rewriter/matchtoterm/TermFieldRewriter.java

```diff
@@ -214,14 +215,23 @@ public boolean isValidIdentifierForTerm(SQLIdentifierExpr expr) {
         return
             !expr.getName().startsWith("_")
             && (
-                (   expr.getParent() instanceof SQLBinaryOpExpr
-                    && ((SQLBinaryOpExpr) expr.getParent()).getOperator() == SQLBinaryOperator.Equality
-                )
-                || expr.getParent() instanceof SQLInListExpr
-                || expr.getParent() instanceof SQLInSubQueryExpr
-                || expr.getParent() instanceof SQLSelectOrderByItem
-                || expr.getParent() instanceof MySqlSelectGroupByExpr
-            );
+                validIdentifierHelper(expr)
+                || (
+                    expr.getParent() instanceof SQLMethodInvokeExpr
+                    && validIdentifierHelper((SQLMethodInvokeExpr) expr.getParent())
+                    )
+               );
+    }
+
+    private boolean validIdentifierHelper(SQLExpr expr) {
+        return
```

Comment:
Why not include `!expr.getName().startsWith("_")` in this method? Seems that's also a requirement for validity.

Reply:
1. This logic is already being checked in `isValidIdentifierForTerm`.
2. It applies only to `SQLIdentifierExpr`.

The whole intention of this function is to de-duplicate code.